tests/thread/enc/CMakeLists.txt

```diff
@@ -43,3 +43,43 @@ enclave_include_directories(pthread_enc PRIVATE ${CMAKE_CURRENT_BINARY_DIR}
 
 enclave_include_directories(oethread_enc PRIVATE ${CMAKE_CURRENT_BINARY_DIR}
                             ${CMAKE_CURRENT_SOURCE_DIR})
+
+add_enclave(
+  TARGET
+  pthread_enc_tcmalloc
+  UUID
+  a499f756-3332-4b4a-ad67-656754c6adbe
+  CXX
+  SOURCES
+  enc.cpp
+  cond_tests.cpp
+  rwlock_tests.cpp
+  errno_tests.cpp
+  thread_t.c)
+
+add_enclave(
+  TARGET
+  oethread_enc_tcmalloc
+  UUID
+  ddd06e41-e7a0-4977-9c3b-b2252bb055c8
+  CXX
+  SOURCES
+  enc.cpp
+  cond_tests.cpp
+  rwlock_tests.cpp
+  errno_tests.cpp
+  thread_t.c)
+
+enclave_compile_definitions(pthread_enc_tcmalloc PRIVATE -D_PTHREAD_ENC_)
+
+enclave_include_directories(
+  pthread_enc_tcmalloc PRIVATE ${CMAKE_CURRENT_BINARY_DIR}
+  ${CMAKE_CURRENT_SOURCE_DIR})
+
+enclave_link_libraries(pthread_enc_tcmalloc oetcmalloc)
+
+enclave_include_directories(
+  oethread_enc_tcmalloc PRIVATE ${CMAKE_CURRENT_BINARY_DIR}
+  ${CMAKE_CURRENT_SOURCE_DIR})
+
+enclave_link_libraries(oethread_enc_tcmalloc oetcmalloc)
```

Comment:
Is the design for user to choose which allocator to use (i.e., specify the library to link)?

Reply:
> Is the design for user to choose which allocator to use (i.e., specify the library to link)?

Yes, the user plugs-in an allocator just be specifying it in the linker line. This is similar to how GBLIC's malloc is replaced.

This PR just locks down the support for tcmalloc; we don't make it available to the user.

Currently, there are two choices in which we can make allocators (snmalloc, tcmalloc) available to the user
- Create an OE SDK sample that shows how to plug in both the allocators (and hence any other allocator that has been ported to work within enclaves).
- Create a separate repository where we show how allocators are plugged in. This has the advantage that more allocators could be supported in the future and could be added without creating more submodule dependencies in OE repository.

In both the above approaches, we just show pluggability; as an advanced feature.